Automattic/jetpack
# Backup: refactor storage level usage to a global state

Related to #28581

## Proposed changes:

* Move storage usage level to a global state, so we can reuse it across different components that will be impacted with p1HpG7-jQY-p2 changes.

### Other information:


## Jetpack product discussion


* p1HpG7-jQY-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






### Unit testing:
You can run unit testing using the following command:
* JS: `jetpack test packages/backup js`

### Testing instructions:
* Navigate to Jetpack -> VaultPress Backup and ensure you are still seeing the storage meter with a black, yellow or green color based on the storage usage.
* You can also check and attempt to change the state using [Redux DevTools (chrome extension)](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en) using this action dispatcher:
```javascript
{
  type: 'SITE_BACKUP_STORAGE_SET',
  usageLevel: 'Full'
}
```
You can use Normal, Warning, Critical or Full on the `usageLevel`. Also, you can play around changing the backup storage size using:
```javascript
{
  type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
  payload: {
    size: 11811160064
  }
}
```
